lib/checkother.cpp

```diff
@@ -1785,7 +1789,7 @@ static bool isConstStatement(const Token *tok, bool cpp)
         return false;
     if (tok->varId() != 0)
         return true;
-    if (Token::Match(tok, "%bool%|%num%|%str%|%char%|nullptr|NULL"))
+    if (Token::Match(tok, constantMatchStr))
```

Comment:
Making it a variable prevents matchcompiling. Same below.

Reply:
Should we make it a #define instead?

Reply:
That's won't help either. The matchcompiler looks for inline strings and "compiles" them. No macro support.

It would help a non-matchcompiled build a bit though.

You could deduplicate the whole function call with a static one instead of just the string.